src/cpp/include/essentiamin.cpp

```diff
@@ -577,6 +577,193 @@ std::vector<float> EssentiaMin::logMelBandsExtractor(std::vector<float>& signal,
 }
 
 
+std::vector<int> EssentiaMin::startStopSilence(std::vector<float>& signal) {
+
+    /////// PARAMS //////////////
+    int frameSize = 1024;
+    int hopSize = 512;
+    int staF = 0;
+    int stoF = 0;
+
+    AlgorithmFactory &factory = standard::AlgorithmFactory::instance();
+    Algorithm* fc    = 	factory.create("FrameCutter",
+                                         "frameSize", frameSize,
+                                         "hopSize", hopSize);
+
+    Algorithm* w       = factory.create("Windowing",
+                                        "type", "hann");
+
+    Algorithm* sss = factory.create("StartStopSilence");
+
+    /////////// CONNECTING THE ALGORITHMS ////////////////
+    std::cout << "-------- connecting algos ---------" << std::endl;
+
+    std::vector<int> startStopFrames;
+    std::vector<Real> frame, windowedFrame;
+
+    fc->input("signal").set(signal);
+    fc->output("frame").set(frame);
+
+    w->input("frame").set(frame);
+    w->output("frame").set(windowedFrame);
+
+    sss->input("frame").set(windowedFrame);
+    sss->output("startFrame").set(staF);
+    sss->output("stopFrame").set(stoF);
+
+    /////////// COMPUTING ////////////////
+    std::cout << "-------- computing ---------" << std::endl;
+
+    while (true) {
+        // compute a frame
+        fc->compute();
+
+        // if it was the last one (ie: it was empty), then we're done.
+        if (!frame.size()) {
+            break;
+        }
+        // if the frame is silent, just drop it and go on processing
+//        if (isSilent(frame)){
+//            cout << "silent frame " << endl;
+//            continue;
+//        }
+        w->compute();
+        sss->compute();
+    }
+
+    delete fc;
+    delete w;
+    delete sss;
+
+    startStopFrames.push_back(staF);
+    startStopFrames.push_back(stoF);
+    return startStopFrames;
+}
+
+//Check https://essentia.upf.edu/documentation/reference/std_GapsDetector.html
+void EssentiaMin::gapsDetector(std::vector<float>& signal, std::vector<float>& starts, std::vector<float>& ends) {
```

Comment:
while `startStopSilence` returns a vector with the results, `gapsDetector` and `discontinuityDetector` write them into separated vectors that are passed as a reference to the function. It'd be better to have consistent behavior. Maybe the easiest way would be to pass the result vector as a reference in `startStopSilence` too.